app/src/db/users.ts

```diff
@@ -277,3 +279,310 @@ export async function updateUserInteraction(
     create: { ...data, userId },
   })
 }
+
+async function getAllCitizens(records: AggregatedType["citizen"]) {
+  return prisma.userAddress.findMany({
+    where: {
+      AND: [
+        {
+          address: {
+            in: records.map((record) => record.address),
+          },
+        },
+      ],
+    },
+    select: {
+      address: true,
+      user: {
+        select: {
+          emails: {
+            select: {
+              email: true,
+            },
+          },
+        },
+      },
+    },
+  })
+}
+
+async function getAllBadgeholders() {
+  return []
+}
+
+async function getAllGovContributors(
+  records: AggregatedType["gov_contribution"],
+) {
+  return prisma.userAddress.findMany({
+    where: {
+      AND: [
+        {
+          address: {
+            in: records.map((record) => record.address),
+          },
+        },
+      ],
+    },
+    select: {
+      address: true,
+      user: {
+        select: {
+          emails: {
+            select: {
+              email: true,
+            },
+          },
+        },
+      },
+    },
+  })
+}
+
+async function getAllRFVoters(records: AggregatedType["rf_voter"]) {
+  return prisma.userAddress.findMany({
+    where: {
+      address: {
+        in: records.map((record) => record.address),
+      },
+    },
+    select: {
+      address: true,
+      user: {
+        select: {
+          emails: {
+            select: {
+              email: true,
+            },
+          },
+        },
+      },
+    },
+  })
+}
+
+async function getAllContributors(records: AggregatedType["contributors"]) {
+  return prisma.project.findMany({
+    where: {
+      id: {
+        in: CONTRIBUTOR_ELIGIBLE_ADDRESSES,
+      },
+    },
+    select: {
+      team: {
+        select: {
+          user: {
+            select: {
+              addresses: {
+                select: {
+                  address: true,
+                },
+              },
+              emails: {
+                select: {
+                  email: true,
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  })
+}
+
+async function getAllOnchainBuilders() {
+  return prisma.user.findMany({
+    where: {
+      OR: [
+        {
+          projects: {
+            some: {
+              project: {
+                contracts: {
+                  some: {
+                    verificationProof: {
+                      not: {
+                        equals: "",
+                      },
+                    },
+                  },
+                },
```

Comment:
this works but checking if the contract exists is enough and more efficient
```suggestion
                  some: { },
                },
```